container/docker/docker.go

```diff
@@ -115,36 +115,46 @@ func Images() ([]v1.DockerImage, error) {
 
 // Checks whether the dockerInfo reflects a valid docker setup, and returns it if it does, or an
 // error otherwise.
-func ValidateInfo() (*dockertypes.Info, error) {
-	client, err := Client()
-	if err != nil {
-		return nil, fmt.Errorf("unable to communicate with docker daemon: %v", err)
-	}
-
-	dockerInfo, err := client.Info(defaultContext())
+func ValidateInfo(GetInfo func() (*dockertypes.Info, error), ServerVersion func() (string, error)) (*dockertypes.Info, error) {
+	info, err := GetInfo()
 	if err != nil {
-		return nil, fmt.Errorf("failed to detect Docker info: %v", err)
+		return nil, err
 	}
 
 	// Fall back to version API if ServerVersion is not set in info.
-	if dockerInfo.ServerVersion == "" {
-		version, err := client.ServerVersion(defaultContext())
+	if info.ServerVersion == "" {
+		var err error
+		info.ServerVersion, err = ServerVersion()
 		if err != nil {
-			return nil, fmt.Errorf("unable to get docker version: %v", err)
+			return nil, fmt.Errorf("unable to get runtime version: %v", err)
```

Comment:
I prefer to call them engines rather then runtimes.  (Prevent confusion with OCI Runtmes like runc and crun).